src/Chainweb/Pact/RestAPI/Server.hs

```diff
@@ -425,12 +428,16 @@ internalPoll cutR cid chain cut requestKeys0 = do
 toPactTx :: Transaction -> Maybe (Command Text)
 toPactTx (Transaction b) = decodeStrict' b
 
-validateCommand :: Command Text -> Either String ChainwebTransaction
-validateCommand cmdText = case verifyCommand cmdBS of
+validateCommand :: ChainwebVersion -> Command Text -> Either String ChainwebTransaction
+validateCommand v cmdText = case verifyCommand cmdBS of
     ProcSucc cmd
-        | length (_cmdSigs cmd) <= 100 -> Right (mkPayloadWithText <$> cmd)
-        | otherwise -> Left "More than 100 signatures given."
+        | length (_cmdSigs cmd) > 100 -> Left "More than 100 signatures given."
+        | payloadVer cmd /= Just v -> Left "Incompatible ChainwebVersion given."
+        | otherwise -> Right (mkPayloadWithText <$> cmd)
     ProcFail err -> Left err
   where
     cmdBS :: Command ByteString
     cmdBS = encodeUtf8 <$> cmdText
+
+    payloadVer :: Command (Payload m c) -> Maybe ChainwebVersion
+    payloadVer = (>>= fromText . Pact._networkId) . _pNetworkId . _cmdPayload
```

Comment:
i have a util in `TransactionExec` that does the first part of this - see `networkIdOf`